build-system/tasks/helpers.js

```diff
@@ -282,7 +282,9 @@ function toEsmName(name) {
 }
 
 function maybeToEsmName(name) {
-  return argv.esm ? toEsmName(name) : name;
+  // If any process is "fortesting" write out the files as non mjs files
+  // since the current system relies too much on looking for "js" files.
```

Comment:
Should we instead fix the system? This seems like a workaround.

Reply:
fair statement, now that we have your html transformer pipeline i would rather use that. im moving the work to #26556 since there were already comments on that one